### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by LibreHacker** [yggdrasil-network/crispa-android#17]
https://gitlab.com/fdroid/rfp/-/issues/new

**Comment by LibreHacker** [comment]
https://gitlab.com/fdroid/rfp/-/issues/1456

**Comment by 6543** [comment]
-> `The original app author has been notified (and supports the inclusion).`

@vikulin @ChronosX88 would it be OK?

the only additional work who had to be done: create changelogs as they are in #23 befor tagging a release and make sure the versionCode iterates 

F-Droid-Bot would do the rest for you after an initial merge-request to https://gitlab.com/fdroid/fdroiddata ( witch I can do ... )

**Comment by ChronosXYZ** [comment]
@6543
> @vikulin @ChronosX88 would it be OK?

Yep, that's ok.

**Comment by LibreHacker** [comment]
See
https://github.com/yggdrasil-network/yggdrasil-android/pull/24

**Comment by 6543** [comment]
I think https://github.com/vikulin/yggdrasil-android/issues/4 is mainly blocking it

**Comment by 6543** [comment]
ok only think left is to create a working f-droid build conf

**Comment by LibreHacker** [comment]
@ChronosX88 whats with https://github.com/yggdrasil-network/yggdrasil-android/blob/master/fastlane/metadata/android/en-US/changelogs/4.txt ?

**Comment by 6543** [comment]
@LibreHacker it look's like v1.4 was just not released jet and so changelogs/4.txt could change/extend ...

**Comment by ChronosXYZ** [comment]
> @ChronosX88 whats with https://github.com/yggdrasil-network/yggdrasil-android/blob/master/fastlane/metadata/android/en-US/changelogs/4.txt ?

@LibreHacker Current latest version is 1.3. 1.4 isn't released yet (and it was not me who added this file, it is @vikulin, idk why he did it).

**Comment by LibreHacker** [comment]
> > @ChronosX88 whats with https://github.com/yggdrasil-network/yggdrasil-android/blob/master/fastlane/metadata/android/en-US/changelogs/4.txt ?
> 
> @LibreHacker Current latest version is 1.3. 1.4 isn't released yet (and it was not me who added this file, it is @vikulin, idk why he did it).

See
https://github.com/yggdrasil-network/yggdrasil-android/pull/22
https://github.com/yggdrasil-network/yggdrasil-android/pull/23
https://gitlab.com/fdroid/rfp/-/issues/1456#note_402714936

You merge pulls needs for support F-Droid after commit v1.3 release.

**Comment by 6543** [comment]
Yes fastlane will have an efect on version >= 1.4 only

**Comment by 6543** [comment]
Witch mean untill a release the app will have no screenshots an description displayed

**Comment by 6543** [comment]
currently blocked https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7319 -> NDK debendency issue on F-Droid build server side

**Comment by 6543** [comment]
@vikulin @ChronosX88 I finaly got a working build config for F-Droid, but it will require to release v1.4 since v1.3 is not compilable for me at all

**Comment by vikulin** [comment]
@6543, could add some more details about exactly what build config changes
are needed?

сб, 29 авг. 2020 г. в 21:19, 6543 <notifications@github.com>:

**Comment by 6543** [comment]
@vikulin just tag :) and start developing v1.5 :)

**Comment by 6543** [comment]
`git tag v1.4; git push -u name-of-upstream-remote --tags`
upstream-remote is mostly called "origin" but you can find it out by `git remote`

**Comment by 6543** [comment]
@vikulin -> https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes

**Comment by 6543** [comment]
@vikulin or via GUI/Github: https://github.com//yggdrasil-network/yggdrasil-android/releases/new

**Comment by ChronosXYZ** [comment]
@6543 yep, we are know how to use the Git, otherwise we wouldn't be sitting here :)

**Comment by vikulin** [comment]
@6543 , I meant have you changed anything in build config? If yes, please share these changes.

**Comment by 6543** [comment]
ah sorry for not getting it :/
yes I have made two changes, both are automaticaly done by f-droid:
```
sed -i '/signingConfig\ signing/d' build.gradle       # signing is done after building by f-droid
sed -i '/ndkVersion/d' build.gradle                   # fixing NDK version missmatch (21.2.x vs 21.3.x)
```

**Comment by vikulin** [comment]
@ChronosX88 , please check the latest 1.4 build from https://t.me/yggdrasil_android . if it's stable enough publish it as 1.4 release in the official repo.

**Comment by ChronosXYZ** [comment]
@vikulin okay, will do

**Comment by 6543** [comment]
blocked by #25 ?

**Comment by ChronosXYZ** [comment]
@6543 I've published the 1.4 version. You can unmark WIP status in [F-Droid new app merge request](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7319).

**Comment by bbigras** [comment]
Any progress on this?

**Comment by 6543** [comment]
at the moment: NO didnt had time for this one :/

**Comment by jl452** [comment]
any news?

**Comment by 6543** [comment]
~~I'll have a look soon :)~~

EDIT: done - have a working build now, just waiting for F-Droid maintainer responce

**Comment by 6543** [comment]
UPDATE: got merged

UPDATE: got build - https://f-droid.org/wiki/index.php?title=io.github.chronosx88.yggdrasil/lastbuild&redirect=yes

- waiting for publishing

**Comment by 6543** [comment]
https://f-droid.org/packages/io.github.chronosx88.yggdrasil/

#17 can be closed 🎉

**Comment by vikulin** [comment]
@6543 Thanks for the work! Closed.

**Comment by bbigras** [comment]
I'm trying it and it seems to be crashing when I enable "ygg me!".

Any easy way to report crash with fdroid app?

**Comment by 6543** [comment]
the fdroid app does not differ from the apk you find at release page, beside the signature ...

so Its a general question: how to send a crash report?

**Comment by bbigras** [comment]
I guess. With most Android apps we can send a report in 1 click (it's probably for google play apps). I was wondering if there was something similar with fdroid.

**Comment by 6543** [comment]
@bbigras no, you have to copy-paste the error into a new issue here, or send it via mail to one of the maintainers ...

**Comment by jl452** [comment]
@bbigras some android apps use some-lib for send crush-reports after crush?

ps: why "This Anti-Feature is applied to apps that promote or depend entirely on a Non-Free network service." ?

**Comment by 6543** [comment]
Anti-Feature; https://github.com/yggdrasil-network/yggdrasil-android/issues/36

**Comment by bbigras** [comment]
> @bbigras some android apps use some-lib for send crush-reports after crush?

https://developer.android.com/games/optimize/crash . It probably depends on google play services.
